### ISSUE TITLE
config setting for database

### ISSUE THREAD

**Issue by ParkSomin23** [adbailey1/DepAudioNet_reproduction#1]
Hi, adbailey1 

I'm interested in your work.

I want to get the same(or similar) results with your work, but I keep getting poor results. Even though I'm using your daic_woz_process.
So I'm wondering that I might be missing some config setting while making the database.
Could I get a response regards to the config setting for making the database, please? Thank you so much for your time for reading.

Here's my config file for making the database. For the DepAudioNet config, I used the same config settings in README.

EXPERIMENT_DETAILS = {'FEATURE_EXP': 'mel', # or 'logmel'
                      'FREQ_BINS': 40,
                      'DATASET_IS_BACKGROUND': False,
                      'WHOLE_TRAIN': False,
                      'WINDOW_SIZE': 1024,
                      'OVERLAP': 50,
                      'SVN': True,
                      'SAMPLE_RATE': 16000,
                      'REMOVE_BACKGROUND': True}

GENDER = False

Best regards,
Somin Park

**Comment by adbailey1** [comment]
Hi Somin, 

Thanks for your message. 

Your config file looks the same as mine. Would you be able to provide more details, such as if you are using the provided conda environment? 

As a test, confirm versions of python=3.7.3, pytorch=1.3.1 Cuda toolkit=10.1.243, cudnn7.6.3, numpy=1.16.4 

For database sanity check, could you test the output of the first mel-svn feature?

import h5py

f1 = h5py.File('path/to/database', 'r')

feat = f1['features'][:]

print(feat[0][0][0])

My output is "-0.023413863"

Could you provide your config file for the actual experiment as well? 

What are the results from your actual experiment? 

Sorry for all the questions but I hope we can get to the bottom of this.

Kind regards, 

Andrew

**Comment by ParkSomin23** [comment]
Hi, adbailey1 

Thank you for your answer. I appreciate it!
While I was checking my config file, I found that my 'EXP_RUNTHROUGH' was set to 1 and had an error related to 'EXP_RUNTHROUGH'(it was my typo problem). It made a big difference to me. 
I'd like to apologize for not being so careful. I really thank you for your time and kind response!

Below I answered your question in case someone needs it.


For me, I used the provided conda environment and my first output of the mel-svn-feature was also "-0.023413863"

My Config file was,

EXPERIMENT_DETAILS = {'FEATURE_EXP': 'mel',
                      'CLASS_WEIGHTS': False,
                      'USE_GENDER_WEIGHTS': False,
                      'SUB_SAMPLE_ND_CLASS': True, 
                      'CROP': True,
                      'OVERSAMPLE': False,
                      'SPLIT_BY_GENDER': True,  # Only for use in test mode
                      'FEATURE_DIMENSIONS': 120,
                      'FREQ_BINS': 40,
                      'BATCH_SIZE': 20,
                      'SVN': True,
                      'LEARNING_RATE': 1e-3,
                      'SEED': 1000,
                      'TOTAL_EPOCHS': 100,
                      'TOTAL_ITERATIONS': 3280,
                      'ITERATION_EPOCH': 1,
                      'SUB_DIR': 'exp_final27_raw1024',
                      'EXP_RUNTHROUGH': 1}

MIN_CROP = True
ANALYSIS_MODE = 'epoch'
WEIGHT_TYPE = 'instance'
GENDER = '-'
WINDOW_SIZE = 1024
HOP_SIZE = 512


My results was,

Results for  'EXP_RUNTHROUGH' = 1:
(I've never got F1 score for D over 0.48)
Scores are: 
[0.7319214603933374, 0.8803571428571424, 0.8803571428571424, 0.8803571428571424, 0.8788156942445511, 0.8801619897317497, 0.8774693987573524, 0.24980860097067698, 0.6213768115942029, 0.8260869565217391, 0.4166666666666667, 0.6258503401360545, 0.7755102040816326, 0.4761904761904762, 0.8865194320678711, 27]

 ###############################################################
train accuracy at epoch: 27
train Accuracy: Mean: 0.88 - [0.88 0.88], F1_Score: Mean: 0.879 - [0.88  0.877], Loss: 0.25
############################################################### 


 ###############################################################
dev accuracy at epoch: 27
dev Accuracy: Mean: 0.621 - [0.826 0.417], F1_Score: Mean: 0.626 - [0.776 0.476], Loss: 0.887
############################################################### 


model_test.log for  'EXP_RUNTHROUGH' = 5:
Scores are: 
[0.6785714285714286, 0.5, 0.8571428571428571, 0.631578947368421, 0.631578947368421, 0.631578947368421, 6.0, 6.0, 1.0, 6.0]
Time taken to evaluate dev: 0.18s
Scores are: 
[0.6090909090909091, 0.8181818181818182, 0.4, 0.6135265700483091, 0.7826086956521738, 0.4444444444444445, 9.0, 2.0, 3.0, 2.0]
Average Scores: 
[ 3.07738095  2.58333333  3.57142857  2.84877258  2.85375089  2.84379427
 31.         29.         10.         25.          2.88181818  4.36363636
  1.4         2.85562457  3.96521739  1.74603175 48.          7.
 18.          7.        ]
Female and Male Confusion Matrices
[['' ' ' '    Prediction' '']
 ['-' '-' '       D' '   ND ']
 ['Ground' 'D ' '0.7142857142857143' '0.2857142857142857']
 ['Truth' 'ND ' '0.48333333333333334' '0.5166666666666667']]
[['' ' ' '    Prediction' '']
 ['-' '-' '       D' '   ND ']
 ['Ground' 'D ' '0.28' '0.72']
 ['Truth' 'ND ' '0.12727272727272726' '0.8727272727272727']]



Best regards,

Somin

**Comment by adbailey1** [comment]
Hi Somin,

No problem at all and no need to apologise, I'm glad you managed to find the error!  :)

Also a quick note, I would change the 'SUB_DIR' name as it currently says 'raw1024' even though the feature you are using is mel. Just thought I would mention as I've made several mistakes due to naming conventions and spent a lot of time trying to work out what certain experiment folders were actually doing! 
 
If you have any other questions or issues please let me know. 

Kind regards, 

Andrew
